src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -606,344 +610,158 @@ public void takeItemStackFromProvider(@NotNull final ICapabilityProvider provide
     }
 
     /**
-     * Wait for a needed shovel.
+     * Wait for a needed tool or weapon.
      *
      * @return NEEDS_SHOVEL
      */
     @NotNull
-    private AIState waitForShovel()
+    private AIState waitForToolOrWeapon()
     {
-        if (checkForShovel())
+        final IToolType toolType = worker.getWorkBuilding().getNeedsTool();
+        if (toolType != ToolType.NONE && checkForToolOrWeapon(toolType))
         {
-            delay += DELAY_RECHECK;
-            return NEEDS_SHOVEL;
+                delay += DELAY_RECHECK;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this line start at column 13. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)